### ISSUE TITLE
transmission-home created within /data folder

### ISSUE THREAD

**Issue by alaorath** [haugene/docker-transmission-openvpn#1521]
**Describe the problem**
Starting a "clean" image results in the transmission home directory (settings.json, logs, etfc) being created within the /data/ path of the image. 

**Expected Result**
The /config/ mount should be used.

**Add your docker run command**
docker run --cap-add=NET_ADMIN -d \
           --name "transmission-VPN" \
           --dns 1.1.1.1 \
           --restart unless-stopped \
              -v /path/to/transmission:/config \
              -v /path/to/transmission/scripts:/scripts \
              -v /path/to/Data:/data \
              -e PGID=666 \
              -e PUID=666 \
              -e OPENVPN_OPTS="--mute-replay-warnings" \
              -e PIA_OPENVPN_CONFIG_BUNDLE=openvpn-ip-nextgen \
              -e OPENVPN_PROVIDER=pia \
              -e OPENVPN_USERNAME=username \
              -e OPENVPN_PASSWORD=password \
              -e TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false \
              -e WEBPROXY_ENABLED=false \
              -e LOCAL_NETWORK=192.168.0.0/24 \
              -e HEALTH_CHECK_HOST=google.com \
              -p 9091:9091 \
              haugene/transmission-openvpn

**Host system:**
QNAP "Container Station" - however I dislike the tools to create Docker containers, so I use docker run for CLI... docker version:
Client:
 Version:      17.09.1-ce
 API version:  1.32
 Go version:   go1.8.3
 Git commit:   0bbe3ac
 Built:        Wed Nov 27 03:52:27 2019
 OS/Arch:      linux/amd64

Server:
 Version:      17.09.1-ce
 API version:  1.32 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   0bbe3ac
 Built:        Wed Nov 27 03:52:27 2019
 OS/Arch:      linux/amd64

**Workaround**
-e TRANSMISSION_HOME=\config

**Comment by haugene** [comment]
Yeah... Don't disagree with you, now. But there's a bit of history here of course. And this is a "very breaking" change.
I mean, how could we roll this change with limited breakage? :thinking:

**Comment by clement-z** [comment]
Yea it's tough...

> I mean, how could we roll this change with limited breakage? 🤔

**If** the change is needed, I guess it would involve some of:

1. At least involve checking whether `TRANSMISSION_HOME` is overridden and if not, add a warning in the logs that the default is going to be changed in the future (and a specific warning if it is not overriden but `/config` **is** mounted).
2. Add "simultaneous" support for both configurations but still default to the old configuration for new containers, even if `/config` is mounted. So look for `/data/settings.json` → `/config/settings.json` → default to `/data` if none exist.
3. Same as above but default to creating `/config/settings.json` instead if none exist, and record the fact that we did that in a file somewhere inside the container's filesystem (so it's specific to the container).
4. And then somehow link all relevant files from `/config` back to `/data` if the flag of step 3 is set (though neither hard nor symbolic links are an option here, so I think it has to be either a lines in `start.sh`/`stop.sh` or script-based, either copying files periodically or using `inotify` to watch for changes).


For steps 2, 3 and 4, I tried to wrap my head around how to make the transition transparent. But it really seems to me that there's very little benefit to it. 

The feature can simply be queued for version 4.0 at which point users who run `latest` and have still not set `TRANSMISSION_HOME` **and** did not mount `/config` will get a descriptive error message on how the default behavior changed.

And if you really want to hold hands, you could add a `migrate` utility entrypoint to perform the transition of config-related files and folders to a new `TRANSMISSION_HOME` (from e.g. a specified `TRANSMISSION_HOME_OLD`) instead of instructing to manually copy the files.

**Comment by asherman07** [comment]
I know this is in the works, but if I define the environment variables is the compose file manually, how will the affect the migration in the future?

below is my compose file 
`transmission-openvpn:
    image: haugene/transmission-openvpn:latest
    container_name: transmission-vpn
    restart: unless-stopped
    cap_add:
        - NET_ADMIN
    volumes:
        - './transmission:/config'
        - '/data/download/torrents:/data/download/torrents'
    environment:
        - TZ=US/Central
        - PUID=1000
        - PGID=998
        - UMASK=000
        - TRANSMISSION_HOME=/config/transmission-home
        - TRANSMISSION_WATCH_DIR=/data/download/torrents/watch
        - TRANSMISSION_DOWNLOAD_DIR=/data/download/torrents
        - TRANSMISSION_INCOMPLETE_DIR=/data/download/torrents
        - OPENVPN_PROVIDER=omitted 
        - OPENVPN_CONFIG=omitted
        - OPENVPN_USERNAME=omitted
        - OPENVPN_PASSWORD=omitted
        - LOCAL_NETWORK=0.0.0.0/24
        # - WEBPROXY_ENABLED=true
        # - WEBPROXY_PORT=8888
        # - WEBPROXY_USERNAME=test
        # - WEBPROXY_PASSWORD=password
    logging:
        driver: json-file
        options:
            max-size: 10m
    ports:
        - '8103:9091'
        # - '8104:8888'
        # - '8105:8102'`

**Comment by pkishino** [comment]
@haugene @clement-z would be nice if we could get this in for 4.0

**Comment by austempest** [comment]
Having a separate /config folder would help a lot. If you are still considering it that is.

**Comment by ksurl** [comment]
I think if it is going to be added, no simultaneous support. Make it very obvious in change log (assuming 4.0) that if you don't want to change your folder, stay on 3.x. and then leave directions for migrating/fixing manually if the migration logic doesn't work. my vote is for `/config` and `/downloads` as default. I'm already using the environment variable to set home to /config and only using /data for downloads

**Comment by pkishino** [comment]
So, I made a simple PR for this.
I checked through the code and simply changing the default in the dockerfile to /config is enough. other locations all use $TRANSMISSION_HOME
I did add a check for the mountpoint and a simple log message stating that default is /config

**Comment by ksurl** [comment]
is it out of scope for this thread to also move out of /data and to /downloads for the download folder?
